### PR TITLE
Created service for get entity types

### PR DESCRIPTION
**Pull request by MattPutnam**

### CODE REVIEW

**Comment by helgridly** [comment]
seems legit

File: src/main/scala/org/broadinstitute/dsde/firecloud/service/EntityService.scala [hunk 1]
```diff
@@ -33,10 +33,45 @@ trait EntityService extends HttpService with FireCloudDirectives {
   private final val ApiPrefix = "entities"
   private implicit val executionContext = actorRefFactory.dispatcher
 
-  val routes = listEntitiesPerTypeRoute
+  val routes = listEntityTypesRoute ~ listEntitiesPerTypeRoute
 
   lazy val log = LoggerFactory.getLogger(getClass)
 
+  @Path("")
```
Comment (rushtong) [inline src/main/scala/org/broadinstitute/dsde/firecloud/service/EntityService.scala]: You can lose this line, the path is inherited from the class level `Api` annotation value property.

File: src/main/scala/org/broadinstitute/dsde/firecloud/service/EntityService.scala [hunk 2]
```diff
@@ -33,10 +33,45 @@ trait EntityService extends HttpService with FireCloudDirectives {
...
+  @ApiOperation(value = "list all entity types in a workspace",
+    nickname = "listEntityTypes",
+    httpMethod = "GET",
+    produces = "application/json",
+    response = classOf[String],
+    responseContainer = "List")
```
Comment (rushtong) [inline src/main/scala/org/broadinstitute/dsde/firecloud/service/EntityService.scala]: I recommend condensing these lines to something more scala-like

File: src/main/scala/org/broadinstitute/dsde/firecloud/service/EntityService.scala [hunk 3]
```diff
@@ -33,10 +33,45 @@ trait EntityService extends HttpService with FireCloudDirectives {
   private final val ApiPrefix = "entities"
```
Comment (rushtong) [inline src/main/scala/org/broadinstitute/dsde/firecloud/service/EntityService.scala]: Not your code, but this is unused, and incorrect if it were.

**Comment by rushtong** [comment]
As noted in hipchat, there are no tests for this service yet. See https://broadinstitute.atlassian.net/browse/DSDEEPB-824

:+1: - looks good. Mostly minor comments. Feel free to address or merge as is.

**Pull request merged**